src/daemon/rpmostreed-os.cxx

```diff
@@ -1138,6 +1140,92 @@ os_handle_get_packages (RPMOSTreeOS *interface, GDBusMethodInvocation *invocatio
   return TRUE;
 }
 
+static void
+query_results_to_builder (HyQuery query, GVariantBuilder *builder, const gchar *id)
+{
+  g_autoptr (GPtrArray) pkglist = NULL;
+  pkglist = hy_query_run (query);
+  for (guint i = 0; i < pkglist->len; i++)
+    {
+      auto pkg = static_cast<DnfPackage *> (g_ptr_array_index (pkglist, i));
+      os_add_package_info_to_builder (pkg, builder, id);
+    }
+}
+
+static void
+search_packages_by_filter (HyQuery query, GVariantBuilder *builder, const gchar *const *names,
+                           int *keynames, const gchar *id)
+{
+  hy_query_clear (query);
+  for (guint j = 0; keynames[j] != -1; j++)
+    {
+      for (guint i = 0; names[i] != NULL; i++)
```

Comment:
This is OK, though it's more common in C to iterate over `char**` by pointer.  Like:

```
for (const char *const*keyname_iter = names; names && *names; names++) {
  const char *keyname = *keyname_iter;
```
or so

Reply:
How does the stop conditions for the loop work? `names && *names`

Reply:
Actually, in this case (the `search_packages_by_filter()` function), wouldn't it be better to use indices to access the contents within the array? The `const gchar *const *names` array is looped over twice since a query is made for both exact and sub string matches of the query term. Incrementing the pointer from the first for loop would mean that it would need to be decremented afterwards before iterating through the second for loop.

Reply:
Note that in my suggestion, we create a new "iterator pointer" each time the `for` loop is entered; we don't change `names`.

In the end these are really the same thing, I bet they compile to the same assembly in general too.  So it's fine as is.